dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -424,6 +427,37 @@ public void OnDateOfBirthSet(DateOnly dateOfBirth)
         DateOfBirth = dateOfBirth;
     }
 
+    public void OnHaveNationalInsuranceNumberSet(bool haveNationalInsuranceNumber)
+    {
+        if (!haveNationalInsuranceNumber)
```

Comment:
If user returns and selects no national insurance number, set NationalInsuranceNumber to null

Reply:
In fact, do we even need HaveNationalInsuranceNumber given it's equivalence to (NationalInsuranceNumber != null)

Reply:
I think it's useful to differentiate between users who have answered the 'Do you have a National Insurance number?' and those who haven't.